docs_user/assemblies/assembly_prepare_director_operator_and_rhoso_for_adoption_process.adoc

```diff
@@ -0,0 +1,424 @@
+export CONTROLLER_SSH
+
+CONTROLLER1_SSH="oc -n $OSPDO_NAMESPACE rsh -c openstackclient openstackclient ssh controller-0.ctlplane"
+export CONTROLLER1_SSH
+CONTROLLER2_SSH=
+export CONTROLLER2_SSH
+CONTROLLER3_SSH=
+export CONTROLLER3_SSH
+
+OSPDO_INTERNAL_API_NET="internalapi"
+export OSPDO_INTERNAL_API_NET
+
+STORAGE_CLASS=$(oc get -n $OSPDO_NAMESPACE pvc openstackclient-hosts -o jsonpath='{.spec.storageClassName}')
+export STORAGE_CLASS
+----
+
+. Get the name of the node where the current {OpenStackShort} Controller virtual machine (VM) is running:
++
+----
+$ CONTROLLER_NODE=$(oc -n $OSPDO_NAMESPACE get vmi -ojson | jq -r '.items[0].status.nodeName')
+export CONTROLLER_NODE
+----
+
+
+. If you are using a combination master/worker cluster with a high availability OSPdO environment,
+configure OSPdO to use a single {OpenStackShort} Controller node:
+
+.. Disable stonith:
++
+----
+$ stonith_resources=$(sudo pcs stonith status|grep Started|awk '{print $2}')
+for in in stonith_resources ;do CONTROLLER1_SSH sudo pcs stonith disable ;done
+----
+.. Put non-Controller nodes in maintenance:
++
+----
+$CONTROLLER1_SSH sudo pcs node maintenance <controller-1> <controller-2>
+----
++
+* Replace `<controller-1>` and `<controller-2>` with the names of the Controller nodes in your environment.
+
+.. Update the quorum for single node:
++
+----
+$CONTROLLER1_SSH sudo systemctl stop corosync
+$CONTROLLER2_SSH sudo systemctl stop corosync
+$CONTROLLER3_SSH sudo systemctl stop corosync
+$CONTROLLER1_SSH sudo pcs quorum update last_man_standing=1 wait_for_all=1
+$CONTROLLER1_SSH sudo systemctl restart corosync
+----
+.. Restart the Pacemaker cluster:
++
+----
+$CONTROLLER1_SSH sudo pcs cluster stop
+$CONTROLLER1_SSH sudo pcs cluster start
+----
+.. Remove the OSPdO node network configuration policies from the other two nodes that are not running the Controller VM by placing a `nodeSelector` on the {OpenShiftShort} node that contains the Controller VM:
++
+----
+$ for i in br-ctlplane br-ex br-osp; do oc patch osnetconfig openstacknetconfig --type json -p '[{"op": "replace", "path": "/spec/attachConfigurations/'$i'/nodeNetworkConfigurationPolicy/nodeSelector", "value": {"kubernetes.io/hostname": "'$CONTROLLER_NODE'"}}]'; done
+----
+
+== Preparing RHOSO for adoption
+
+When deploying a RHOSO env. usign the same master nodes, make sure
+to take the following steps:
+
+Get the names of the other two nodes that we will use for NG
+[,bash]
+----
+NG_NODES=$(oc get nodes -o name | grep -v $CONTROLLER_NODE | sed 's#node/##g' | tr '\n' ' ')
+----
+Extract nodes to be used for OSP 18 installation and label nodes.
+[,bash]
+----
+OSP18_NODE1=$(echo "${NG_NODES}" | cut -d ' ' -f 1)
+export OSP18_NODE1
+OSP18_NODE2=$(echo "${NG_NODES}" | cut -d ' ' -f 2)
+export OSP18_NODE2
+oc label nodes "${OSP18_NODE1}" type=openstack
+oc label nodes "${OSP18_NODE2}" type=openstack
+----
```

Comment:
```suggestion
. Extract the nodes to use for OpenStack 18 installation and label the nodes:
+
----
$ OSP18_NODE1=$(echo "${NG_NODES}" | cut -d ' ' -f 1)
export OSP18_NODE1
$ OSP18_NODE2=$(echo "${NG_NODES}" | cut -d ' ' -f 2)
export OSP18_NODE2
$ oc label nodes "${OSP18_NODE1}" type=openstack
$ oc label nodes "${OSP18_NODE2}" type=openstack
----
```
@pinikomarov Is OpenStack 18 correct? It shouldn't be RHOSO 18.0 instead?

Reply:
right changing